backend/app/util.js

```diff
@@ -120,31 +122,29 @@ exports.Query = function (realm) {
             return rows;
         }
 
-        function doQuery(queryObject, client, done, options, cb) {
-            var queryString;
+        function doQuery(queryObject, options, cb) {
             if (typeof queryObject === 'string') {
-
-                queryString =
-                    (typeof realm !== 'undefined') ?
-                    ('SET search_path TO ' + realm + '; ' + queryObject) : queryObject;
-                debug(queryString);
-
-                client.query(queryString, options, function (err, result) {
-                    done();
-                    var cbfunc = (typeof cb === 'function');
-
+                if (typeof realm !== 'undefined') {
+                    debug('SET search_path TO ' + realm);
+                    pool.query('SET search_path TO ' + realm)
+                        .catch(setErr => {
```

Comment:
You cannot leave a promise dangling like this when there are a callback argument and other promises in the same functions. I would move the first query to a function like      function a () {
        if (typeof realm !== 'undefined') {
             return pool.query('SET search_path TO ' + realm)
        } else {
            return Promise.resolve();
        }
     }
and chain it before the  second query.